distributed/config.py

```diff
@@ -24,8 +24,13 @@ def ensure_config_file(destination=dask_config_path):
                     os.mkdir(os.path.dirname(destination))
                 except FileExistsError:
                     pass
-            shutil.copy(default_path, destination)
-
+            # Atomically create destination
+            tmp = '%s.tmp.%d' % (destination, os.getpid())
+            shutil.copy(default_path, tmp)
```

Comment:
Might be nice to add a comment explaining why we do this instead of just copying directly.